src/widgets/panel/panel.js

```diff
@@ -0,0 +1,145 @@
+import { component } from '../../lib/suit';
+import Template from '../../components/Template/Template';
+
+const suit = component('Panel');
+
+const renderer = ({ containerNode, cssClasses, templates, templateProps }) => ({
+  renderingOptions,
+  hidden,
+}) => {
+  let bodyReference = null;
+
+  render(
+    <div
+      className={cx(cssClasses.root, {
+        [cssClasses.noRefinementRoot]: !renderingOptions.canRefine,
+      })}
+      hidden={hidden}
+    >
+      {templates.header && (
+        <Template
+          {...templateProps}
+          templateKey="header"
+          rootProps={{
+            className: cssClasses.header,
+          }}
+          data={renderingOptions}
+        />
+      )}
+      <div className={cssClasses.body} ref={ref => (bodyReference = ref)} />
+      {templates.footer && (
+        <Template
+          {...templateProps}
+          templateKey="footer"
+          rootProps={{
+            className: cssClasses.footer,
+          }}
+          data={renderingOptions}
+        />
+      )}
+    </div>,
+    containerNode
+  );
+
+  return { bodyReference };
+};
+
+renderer.propTypes = {
+  templates: PropTypes.object.isRequired,
+  templateProps: PropTypes.object.isRequired,
+  cssClasses: PropTypes.shape({
+    root: PropTypes.string.isRequired,
+    noRefinementRoot: PropTypes.string.isRequired,
+    body: PropTypes.string.isRequired,
+    header: PropTypes.string.isRequired,
+    footer: PropTypes.string.isRequired,
+  }).isRequired,
+};
+
+const usage = `Usage:
+const widgetWithHeaderFooter = panel({
+  container,
+  [ cssClasses.{root, noRefinementRoot, body, header, footer} ],
+  [ templates.{header, footer} ],
+})(widget);
+
+const myWidget = widgetWithHeaderFooter(widgetOptions)`;
+
+export default function panel({
+  templates,
+  hidden = () => false,
+  cssClasses: userCssClasses = {},
+} = {}) {
+  const cssClasses = {
+    root: cx(suit(), userCssClasses.root),
+    noRefinementRoot: cx(
+      suit({ modifierName: 'noRefinement' }),
+      userCssClasses.noRefinementRoot
+    ),
+    body: cx(suit({ descendantName: 'body' }), userCssClasses.body),
+    header: cx(suit({ descendantName: 'header' }), userCssClasses.header),
+    footer: cx(suit({ descendantName: 'footer' }), userCssClasses.footer),
+  };
+
+  return widgetFactory => widgetOptions => {
+    const { container } = widgetOptions;
+
+    if (!container) {
+      throw new Error(
+        `[InstantSearch.js] The widget "${
+          widgetFactory.name
+        }" is missing a container option.`
+      );
+    }
+
+    const defaultTemplates = { header: '', footer: '' };
+    const templateProps = prepareTemplateProps({ defaultTemplates, templates });
+
+    const renderPanel = renderer({
+      containerNode: getContainerNode(container),
+      cssClasses,
+      templates,
+      templateProps,
+    });
+
+    try {
+      const { bodyReference } = renderPanel({
+        renderingOptions: {
+          canRefine: false,
```

Comment:
I'm not sure if the default value should be `true` or `false`.

If ever some developer wants to add an fade-out animation when there's no refinement possible, it would flash on load with a default value set to `false`.

Reply:
Yep! Let's experiment a little on our side and see what makes more sense.

Keep in mind that if we were to change the default value, it would be a behavior change (and potentially a breaking change).